src/Staking/StakingCenter/StakingCenter.tsx

```diff
@@ -1,172 +1,90 @@
+import {useStakingTx} from '../../Dashboard/StakePoolInfos'
 import {useLanguage} from '../../i18n'
 import globalMessages, {errorMessages} from '../../i18n/global-messages'
 import {showErrorDialog} from '../../legacy/actions'
-import {
-  CONFIG,
-  getDefaultAssetByNetworkId,
-  getTestStakingPool,
-  isNightly,
-  SHOW_PROD_POOLS_IN_DEV,
-} from '../../legacy/config'
-import {ApiError, NetworkError} from '../../legacy/errors'
-import {normalizeTokenAmount} from '../../legacy/format'
+import {CONFIG, getTestStakingPool, isNightly, SHOW_PROD_POOLS_IN_DEV} from '../../legacy/config'
 import {Logger} from '../../legacy/logging'
 import {getNetworkConfigById} from '../../legacy/networks'
-import {
-  accountBalanceSelector,
-  isFetchingUtxosSelector,
-  poolOperatorSelector,
-  utxosSelector,
-} from '../../legacy/selectors'
-import {isEmptyString} from '../../legacy/utils'
 import {StakingCenterRouteNavigation} from '../../navigation'
 import {useSelectedWallet} from '../../SelectedWallet'
-import {UtxoAutoRefresher} from '../../UtxoAutoRefresher'
-import {NotEnoughMoneyToSendError, YoroiWallet} from '../../yoroi-wallets'
-import {DefaultAsset} from '../../yoroi-wallets/types'
-import {RawUtxo} from '../../yoroi-wallets/types/other'
+import {NotEnoughMoneyToSendError} from '../../yoroi-wallets'
 import {PoolDetailScreen} from '../PoolDetails'
-import {PoolWarningModal} from '../PoolWarningModal'
 
 export const StakingCenter = () => {
   const intl = useIntl()
   const navigation = useNavigation<StakingCenterRouteNavigation>()
-  const [amountToDelegate, setAmountToDelegate] = useState<string | null>(null)
-  const [selectedPools, setSelectedPools] = useState<Array<SelectedPool>>([])
-  const [reputationInfo, setReputationInfo] = useState({})
-  const [showPoolWarning, setShowPoolWarning] = useState(false)
-  const [busy, setBusy] = useState(false)
 
-  const isFetchingUtxos = useSelector(isFetchingUtxosSelector)
-  const utxos = useSelector(utxosSelector)
-  const accountBalance = useSelector(accountBalanceSelector)
-  const poolOperator = useSelector(poolOperatorSelector)
   const {languageCode} = useLanguage()
   const wallet = useSelectedWallet()
-  const config = getNetworkConfigById(wallet.networkId)
-  const isMainnet = config.IS_MAINNET
-
-  const nightlyAndDevPoolHashes = getTestStakingPool(wallet.networkId, wallet.provider)
 
-  // pools user is currently delegating to
-  const poolList = poolOperator != null ? [poolOperator] : null
-
-  const handleOnPress = (poolHash: string) => {
-    const selectedPoolHashes = !isEmptyString(poolHash) ? [poolHash] : nightlyAndDevPoolHashes
-    Logger.debug('manual inputted or config pool:', selectedPoolHashes)
-    delegate(selectedPoolHashes)
-  }
+  const [selectedPoolId, setSelectedPoolId] = React.useState<string>()
+
+  const {isLoading} = useStakingTx(
+    {wallet, poolId: selectedPoolId},
+    {
+      onSuccess: (yoroiUnsignedTx) => {
+        if (selectedPoolId == null) return
```

Comment:
if the `poolId` is null it will always trigger `onError` 🤔 , but the query is only enabled if pool !== null. I think we can delete this line.

Reply:
it's a typescript error